src/app/_components/product.tsx

```diff
@@ -0,0 +1,221 @@
+"use client";
+import { Switch } from "../_components/ui/switch";
+import React, { useState } from "react";
+import { Button, buttonVariants } from "../_components/ui/button";
+import Image from "next/image";
+
+interface Product {
+  id: number;
+  name: string;
+  description: string | null;
+  calification: string | null;
+  personalize: string[];
+  price: number;
+  restaurant: string | null;
+  imageUrl?: string | null;
+}
+
+interface ProductProps {
+  product: Product;
+  variant?: "default" | "mobile";
+}
+
+const Product = ({ product, variant = "default" }: ProductProps) => {
+  const { name, description, calification, personalize, price, restaurant } =
+    product || {};
+
+  const [quantity, setQuantity] = useState(0);
+
+  const decreaseQuantity = () => {
+    setQuantity((prevQuantity) => (prevQuantity > 0 ? prevQuantity - 1 : 0));
+  };
+
+  const increaseQuantity = () => {
+    setQuantity((prevQuantity) => prevQuantity + 1);
+  };
+
+  const descriptionItems =
+    description?.split("\n").filter((item) => item.trim() !== "") ?? [];
+  const [firstItem, ...restItems] = descriptionItems;
```

Comment:
From what I discussed with Facu, seems like he also likes the standard of: "if an instruction uses more than one line, then we separate it from the other ones with blank lines", so I guess it's kind of becoming a standard in the company, so let's keep that in mind. With that mentality a blank line would be needed between this two instructions